tests/integration/components/feature-controls-test.js

```diff
@@ -94,4 +118,52 @@ module('Integration | Component | feature-controls', function(hooks) {
       '❗'
     );
   });
+
+  test('pressing the reset button rolls back to default state', async function(assert) {
+    this.setProperties({
+      features: featuresMock(assert),
+      featureFlags
+    });
+    await render(
+      hbs`{{feature-controls features=features featureFlags=featureFlags}}`
+    );
+    await click('[data-test-checkbox-flag="flagTrue"]');
+    assert.notOk(
+      this.element.querySelector('[data-test-checkbox-flag="flagTrue"]').checked
+    );
+    await click('[data-test-checkbox-flag="flagFalse"]');
+    assert.ok(
+      this.element.querySelector('[data-test-checkbox-flag="flagFalse"]')
+        .checked
+    );
+
+    await click('[data-test-button-reset]');
+    assert.ok(
+      this.element.querySelector('[data-test-checkbox-flag="flagTrue"]').checked
+    );
+    assert.notOk(
+      this.element.querySelector('[data-test-checkbox-flag="flagFalse"]')
+        .checked
+    );
+  });
+
+  test('pressing refresh updates the model', async function(assert) {
```

Comment:
For this test, you can update the property featureFlags of the component to some the "default data" so the refresh function will reuse the updated data ;)

In the component we will never override featureFlags attibute, because it's the listing of the config/environnement file :)

Reply:
@GreatWizard thanks for the great suggestion